django_advanced_queries/covid_19/tests.py

```diff
@@ -429,76 +469,110 @@ def test_detect_potential_infected_patients_because_of_sick_hospital_worker(self
 
         # Now improve the test to hit DB once only
         with self.assertNumQueries(1):
-            patient_examined_by_sick_hospital_worker = Patient.objects.filter_by_examined_hospital_workers(
-                hospital_workers=NotImplementedError
-            )
-            num_of_patient_examined_by_sick_hospital_worker = patient_examined_by_sick_hospital_worker.count()
+            results_subquery = MedicalExaminationResult.objects.filter(
+                patient__person=OuterRef(
+                    'medical_examination_results__examined_by')).order_by(
+                '-time')
+
+            patient_examined_by_sick_hospital_worker = Patient.objects.annotate(
+                last_result=Subquery(
+                    results_subquery.values('result')[:1])).exclude(
+                last_result__in=[MedicalExaminationResult.RESULT_DEAD,
+                                 MedicalExaminationResult.RESULT_HEALTHY]).distinct()
+
+            num_of_patient_examined_by_sick_hospital_worker = len(
+                list(patient_examined_by_sick_hospital_worker))
```

Comment:
list is redundant, `len` will evaluate it

Reply:
d